packages/nouns-contracts/contracts/NounsArt.sol

```diff
@@ -0,0 +1,453 @@
+     */
+    function glassesPage(uint256 pageIndex) public view override returns (INounsArt.NounArtStoragePage memory) {
+        return _glasses.storagePages[pageIndex];
+    }
+
+    /**
+     * @notice Get a head image bytes (RLE-encoded).
+     */
+    function heads(uint256 index) public view override returns (bytes memory) {
+        return imageByIndex(_heads, index);
+    }
+
+    /**
+     * @notice Get a body image bytes (RLE-encoded).
+     */
+    function bodies(uint256 index) public view override returns (bytes memory) {
+        return imageByIndex(_bodies, index);
+    }
+
+    /**
+     * @notice Get a accessory image bytes (RLE-encoded).
+     */
+    function accessories(uint256 index) public view override returns (bytes memory) {
+        return imageByIndex(_accessories, index);
+    }
+
+    /**
+     * @notice Get a glasses image bytes (RLE-encoded).
+     */
+    function glasses(uint256 index) public view override returns (bytes memory) {
+        return imageByIndex(_glasses, index);
+    }
+
+    /**
+     * @notice Get a color palette bytes.
+     */
+    function palettes(uint8 paletteIndex) public view override returns (bytes memory) {
+        return SSTORE2.read(palettesPointers[paletteIndex]);
+    }
+
+    function _addBackground(string calldata _background) internal {
+        backgrounds.push(_background);
+    }
+
+    function addToTraitFromBytes(
+        Trait storage trait,
+        bytes calldata encodedCompressed,
+        uint80 decompressedLength,
+        uint16 imageCount
+    ) internal {
+        if (encodedCompressed.length == 0) {
+            revert EmptyBytes();
+        }
+        address pointer = SSTORE2.write(encodedCompressed);
+        addToTraitFromPointer(trait, pointer, decompressedLength, imageCount);
+    }
+
+    function addToTraitFromPointer(
+        Trait storage trait,
+        address pointer,
+        uint80 decompressedLength,
+        uint16 imageCount
+    ) internal {
+        if (decompressedLength == 0) {
+            revert BadDecompressedLength();
+        }
+        if (imageCount == 0) {
+            revert BadImageCount();
+        }
+        trait.storagePages.push(
+            NounArtStoragePage({ pointer: pointer, decompressedLength: decompressedLength, imageCount: imageCount })
+        );
+        trait.storedImagesCount += imageCount;
+    }
+
+    function imageByIndex(INounsArt.Trait storage trait, uint256 index) internal view returns (bytes memory) {
+        (INounsArt.NounArtStoragePage storage page, uint256 indexInPage) = getPage(trait.storagePages, index);
+        bytes[] memory decompressedImages = decompressAndDecode(page);
+        return decompressedImages[indexInPage];
+    }
+
+    function getPage(INounsArt.NounArtStoragePage[] storage pages, uint256 index)
+        internal
+        view
+        returns (INounsArt.NounArtStoragePage storage, uint256)
+    {
+        uint256 len = pages.length;
+        if (len == 0) {
```

Comment:
I think this check can be dropped, and it will revert on `ImageNotFound()` which I think makes sense

Reply:
https://github.com/nounsDAO/nouns-monorepo/pull/499/commits/15d27b69c1844d5d017747d4acc3745e6a5f289b